From 4b5657317e4f26a386531917b2d981207534f0cb Mon Sep 17 00:00:00 2001
From: Yu Zhao <yuzhao@google.com>
Date: Mon, 10 May 2021 13:09:21 -0600
Subject: [PATCH 19/19] mm/vmscan.c: fix warnings from walk_pud_range()

Correctly skip unevictable VMAs that contain special PFNs. Otherwise
those PFNs trigger a warning in walk_pud_range(), which is harmless
but noisy, when using multigenerational LRU.
---
 mm/vmscan.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/mm/vmscan.c b/mm/vmscan.c
index 7e10e8bb6..cd4f1c6ad 100644
--- a/mm/vmscan.c
+++ b/mm/vmscan.c
@@ -4997,8 +4997,7 @@ static bool get_next_interval(struct mm_walk *walk, unsigned long mask, unsigned
 		if ((next & mask) != (walk->vma->vm_start & mask))
 			return false;
 
-		if (next <= walk->vma->vm_start &&
-		    should_skip_vma(walk->vma->vm_start, walk->vma->vm_end, walk)) {
+		if (should_skip_vma(walk->vma->vm_start, walk->vma->vm_end, walk)) {
 			walk->vma = walk->vma->vm_next;
 			continue;
 		}
-- 
2.31.1.527.g2d677e5b15

